src/debugger/launch.ts

```diff
@@ -35,6 +37,11 @@ export async function makeDebugConfigurations(ctx: FolderContext) {
     for (const config of configs) {
         const index = launchConfigs.findIndex(c => c.name === config.name);
         if (index !== -1) {
+            if (regenerate && launchConfigs[index] !== config) {
+                launchConfigs[index] = config;
```

Comment:
It's hard to actually prevent us from covering user edits because they're also free to edit the paths themselves... And the logic/consequences can be hard to describe.

To regenerate here makes the logic general and applicable to other cases that may change the base config. I intended to notify user about the risk of their config being covered by using "Regenerate" instead of "Update" here.

Reply:
I find this hard to implement because it's hard to update nested value in TypeScript🤷‍♂️definitely missing key path in Swift.